Submissions/Kademgari_Week 3 - Functions and Objects.playground/Contents.swift

```diff
@@ -0,0 +1,223 @@
+//: ## Functions and Closures
+//: 1.)
+//: Remove the `day` parameter. Add a parameter to include today’s lunch special in the greeting.
+//:
+//:  Write a custom argument label before the parameter name, or write `_` to use no argument label.
+//:
+func greet(_ person: String, lunchspecial: String) -> String {
+    return "Hello \(person), today's lunch special is \(lunchspecial)."
+}
+greet("John", lunchspecial: "Noodles")
+
+//: 2.)
+//:
+//: Rewrite the closure to return zero for all odd numbers.
+//:
+var numbers = [20, 19, 7, 12]
+
+//: full "syntactic sugar" form:
+numbers.map({ (number: Int) -> Int in
+    if(number % 2 != 0)
+    {
+        return 0
+        
+    }
+    else
+    {
+        let result = 3 * number
+        return result
+        
+    }
+})
+//: omitting parameters and return type form:
+let mappedNumbers = numbers.map({ number in 3 * number })
+print(mappedNumbers)
+//: reference variable form:
+let mappedNums = numbers.map { 3 * $0 % 2 == 0 ? 3 * $0 :0}
+print(mappedNums)
+//: ## Objects and Classes
+//: 3.)
+//:
+//: Add a constant property with `let`, and add another method that takes an argument.
+//:
+//: 4.)
+//:
+//: This version of the `Shape` class is missing something important: an initializer to set up the class when an instance is created. Use `init` to create one.
+//:
+class Shape {
+    var numberOfSides = 0
+    let myvariable = 4
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+    func argpassing(dimensions:Int) -> Bool {
+        var result = true
+        if(myvariable == dimensions){
+            result = true
+        }
+        return result
+        }
+    init(count: Int){
+        numberOfSides = count
+    }
+}
+var testclass = Shape(count:4)
+testclass.argpassing(dimensions: 4)
+testclass.numberOfSides = 2
+
+//: 5.)
+//:
+//: Make a subclass of `NamedShape` called `Circle` that takes a radius and a name as arguments to its initializer. Implement an `area()` and a `simpleDescription()` method on the `Circle` class.
+//:
+class NamedShape {
+    var numberOfSides: Int = 0
+    var name: String
+    
+    init(name: String) {
+        self.name = name
+    }
+    
+    func simpleDescription() -> String {
+        return "A shape with \(numberOfSides) sides."
+    }
+}
+class Circle:NamedShape{
```

Comment:
include spacing around your punctuation... after your colons and commas, before your curly bracket, a blank line between variables and functions to let the eye rest and read the code easily. :) Whitespace takes no memory, so you're not doing any harm including it.